python/cog/server/helpers.py

```diff
@@ -148,3 +162,98 @@ def run(self) -> None:
                     if drain_tokens_seen >= drain_tokens_needed:
                         self.drain_event.set()
                         drain_tokens_seen = 0
+
+
+X = TypeVar("X")
+Y = TypeVar("Y")
+Coro = Coroutine[None, None, Any]
+
+
+@overload
+async def select(
+    x: Coroutine[None, None, X],
+    y: Coroutine[None, None, Y],
+    *,
+    timeout: Optional[float] = None,
+) -> Union[X, Y]:
+    ...
+
+
+@overload
+async def select(*awaitables: Coro, timeout: Optional[float] = None) -> Any:
+    ...
+
+
+async def select(
+    x: Coro, y: Coro, *awaitables: Coro, timeout: Optional[float] = None
```

Comment:
```suggestion
    x: Coro, y: Coro, *awaitables: List[Coro], timeout: Optional[float] = None
```
?

Reply:
not how variadics work, the list is implicit (also it would be a tuple and not a list)

Reply:
you can tell because the typechecker passes